tensorflow/core/kernels/candidate_sampler_ops.cc

```diff
@@ -134,6 +134,7 @@ class AllCandidateSamplerOp : public BaseCandidateSamplerOp {
       : BaseCandidateSamplerOp(context) {
     int64 range_max;
     OP_REQUIRES_OK(context, context->GetAttr("num_sampled", &range_max));
+    CHECK_LE(num_sampled_, range_max);
```

Comment:
Can you use OP_REQUIRES(ctx, num_sampled_, range_max, errors::InvalidArgument("...", )) instead? 

It'll also better to add a test case in the corresponding python test to check the expected exception is raised.